File: sdks/thunkmetrc-cs/ThunkMetrc.Wrapper/Models/TransferTemplateDeliveryPackage.cs
using System;
using System.Collections.Generic;

namespace ThunkMetrc.Wrapper.Models
{
    public class TransferTemplateDeliveryPackage
    {
        [global::System.Text.Json.Serialization.JsonPropertyName("ContainsRemediatedProduct")]
        public required bool ContainsRemediatedProduct { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ExpirationDate")]
        public string? ExpirationDate { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ExternalId")]
        public int? ExternalId { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("GrossUnitOfWeightName")]
        public string? GrossUnitOfWeightName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("IsDonation")]
        public required bool IsDonation { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("IsTestingSample")]
        public required bool IsTestingSample { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("IsTradeSample")]
        public required bool IsTradeSample { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("IsTradeSamplePersistent")]
        public required bool IsTradeSamplePersistent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemBrandName")]
        public string? ItemBrandName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemCategoryName")]
        public required string ItemCategoryName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemId")]
        public required int ItemId { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemName")]
        public required string ItemName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemServingSize")]
        public string? ItemServingSize { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemStrainName")]
        public string? ItemStrainName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemSupplyDurationDays")]
        public int? ItemSupplyDurationDays { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdAContent")]
        public double? ItemUnitCbdAContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdAContentDose")]
        public double? ItemUnitCbdAContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdAContentDoseUnitOfMeasureName")]
        public string? ItemUnitCbdAContentDoseUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdAContentUnitOfMeasureName")]
        public string? ItemUnitCbdAContentUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdAPercent")]
        public double? ItemUnitCbdAPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdContent")]
        public double? ItemUnitCbdContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdContentDose")]
        public double? ItemUnitCbdContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdContentDoseUnitOfMeasureName")]
        public string? ItemUnitCbdContentDoseUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdContentUnitOfMeasureName")]
        public string? ItemUnitCbdContentUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitCbdPercent")]
        public double? ItemUnitCbdPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitQuantity")]
        public double? ItemUnitQuantity { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitQuantityUnitOfMeasureName")]
        public string? ItemUnitQuantityUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcAContent")]
        public double? ItemUnitThcAContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcAContentDose")]
        public double? ItemUnitThcAContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcAContentDoseUnitOfMeasureName")]
        public string? ItemUnitThcAContentDoseUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcAContentUnitOfMeasureName")]
        public string? ItemUnitThcAContentUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcAPercent")]
        public double? ItemUnitThcAPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcContent")]
        public double? ItemUnitThcContent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcContentDose")]
        public double? ItemUnitThcContentDose { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcContentDoseUnitOfMeasureName")]
        public string? ItemUnitThcContentDoseUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcContentUnitOfMeasureName")]
        public string? ItemUnitThcContentUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitThcPercent")]
        public double? ItemUnitThcPercent { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitVolume")]
        public double? ItemUnitVolume { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitVolumeUnitOfMeasureName")]
        public string? ItemUnitVolumeUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitWeight")]
        public double? ItemUnitWeight { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ItemUnitWeightUnitOfMeasureName")]
        public string? ItemUnitWeightUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("LabTestingState")]
        public required string LabTestingState { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("PackageId")]
        public required int PackageId { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("PackageLabel")]
        public required string PackageLabel { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("PackageType")]
        public required string PackageType { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("PackagedDate")]
        public string? PackagedDate { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductLabel")]
        public required TransfersTransferTemplateDeliveryPackageProductLabel ProductLabel { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductRequiresRemediation")]
        public required bool ProductRequiresRemediation { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ProductionBatchNumber")]
        public string? ProductionBatchNumber { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ReceivedQuantity")]
        public required double ReceivedQuantity { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ReceivedUnitOfMeasureName")]
        public required string ReceivedUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("RemediationDate")]
        public string? RemediationDate { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SellByDate")]
        public string? SellByDate { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ShipmentPackageState")]
        public required string ShipmentPackageState { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ShippedQuantity")]
        public required double ShippedQuantity { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("ShippedUnitOfMeasureName")]
        public required string ShippedUnitOfMeasureName { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SourceHarvestNames")]
        public string? SourceHarvestNames { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SourcePackageIsDonation")]
        public required bool SourcePackageIsDonation { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SourcePackageIsTradeSample")]
        public required bool SourcePackageIsTradeSample { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SourcePackageLabels")]
        public string? SourcePackageLabels { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("SourceProductionBatchNumbers")]
        public string? SourceProductionBatchNumbers { get; set; }
        [global::System.Text.Json.Serialization.JsonPropertyName("UseByDate")]
        public string? UseByDate { get; set; }
        public class TransfersTransferTemplateDeliveryPackageProductLabel
        {
            [global::System.Text.Json.Serialization.JsonPropertyName("IsActive")]
            public required bool IsActive { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("IsChildFromParentWithLabel")]
            public required bool IsChildFromParentWithLabel { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("IsDiscontinued")]
            public required bool IsDiscontinued { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("LastLabelGenerationDate")]
            public string? LastLabelGenerationDate { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("PackageId")]
            public int? PackageId { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("QrCount")]
            public required int QrCount { get; set; }
            [global::System.Text.Json.Serialization.JsonPropertyName("TagId")]
            public int? TagId { get; set; }
        }
    }
}

File: sdks/thunkmetrc-rs/wrapper/src/models/transfer_template_delivery_package.rs
use serde::{Deserialize, Serialize};
#[allow(unused_imports)]
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TransferTemplateDeliveryPackage {
    #[serde(rename = "ContainsRemediatedProduct", skip_serializing_if = "Option::is_none")]
    pub contains_remediated_product: Option<bool>,
    #[serde(rename = "ExpirationDate", skip_serializing_if = "Option::is_none")]
    pub expiration_date: Option<String>,
    #[serde(rename = "ExternalId", skip_serializing_if = "Option::is_none")]
    pub external_id: Option<i64>,
    #[serde(rename = "GrossUnitOfWeightName", skip_serializing_if = "Option::is_none")]
    pub gross_unit_of_weight_name: Option<String>,
    #[serde(rename = "IsDonation", skip_serializing_if = "Option::is_none")]
    pub is_donation: Option<bool>,
    #[serde(rename = "IsTestingSample", skip_serializing_if = "Option::is_none")]
    pub is_testing_sample: Option<bool>,
    #[serde(rename = "IsTradeSample", skip_serializing_if = "Option::is_none")]
    pub is_trade_sample: Option<bool>,
    #[serde(rename = "IsTradeSamplePersistent", skip_serializing_if = "Option::is_none")]
    pub is_trade_sample_persistent: Option<bool>,
    #[serde(rename = "ItemBrandName", skip_serializing_if = "Option::is_none")]
    pub item_brand_name: Option<String>,
    #[serde(rename = "ItemCategoryName", skip_serializing_if = "Option::is_none")]
    pub item_category_name: Option<String>,
    #[serde(rename = "ItemId", skip_serializing_if = "Option::is_none")]
    pub item_id: Option<i64>,
    #[serde(rename = "ItemName", skip_serializing_if = "Option::is_none")]
    pub item_name: Option<String>,
    #[serde(rename = "ItemServingSize", skip_serializing_if = "Option::is_none")]
    pub item_serving_size: Option<String>,
    #[serde(rename = "ItemStrainName", skip_serializing_if = "Option::is_none")]
    pub item_strain_name: Option<String>,
    #[serde(rename = "ItemSupplyDurationDays", skip_serializing_if = "Option::is_none")]
    pub item_supply_duration_days: Option<i64>,
    #[serde(rename = "ItemUnitCbdAContent", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_a_content: Option<f64>,
    #[serde(rename = "ItemUnitCbdAContentDose", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_a_content_dose: Option<f64>,
    #[serde(rename = "ItemUnitCbdAContentDoseUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_a_content_dose_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitCbdAContentUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_a_content_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitCbdAPercent", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_a_percent: Option<f64>,
    #[serde(rename = "ItemUnitCbdContent", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_content: Option<f64>,
    #[serde(rename = "ItemUnitCbdContentDose", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_content_dose: Option<f64>,
    #[serde(rename = "ItemUnitCbdContentDoseUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_content_dose_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitCbdContentUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_content_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitCbdPercent", skip_serializing_if = "Option::is_none")]
    pub item_unit_cbd_percent: Option<f64>,
    #[serde(rename = "ItemUnitQuantity", skip_serializing_if = "Option::is_none")]
    pub item_unit_quantity: Option<f64>,
    #[serde(rename = "ItemUnitQuantityUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_quantity_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitThcAContent", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_a_content: Option<f64>,
    #[serde(rename = "ItemUnitThcAContentDose", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_a_content_dose: Option<f64>,
    #[serde(rename = "ItemUnitThcAContentDoseUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_a_content_dose_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitThcAContentUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_a_content_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitThcAPercent", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_a_percent: Option<f64>,
    #[serde(rename = "ItemUnitThcContent", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_content: Option<f64>,
    #[serde(rename = "ItemUnitThcContentDose", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_content_dose: Option<f64>,
    #[serde(rename = "ItemUnitThcContentDoseUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_content_dose_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitThcContentUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_content_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitThcPercent", skip_serializing_if = "Option::is_none")]
    pub item_unit_thc_percent: Option<f64>,
    #[serde(rename = "ItemUnitVolume", skip_serializing_if = "Option::is_none")]
    pub item_unit_volume: Option<f64>,
    #[serde(rename = "ItemUnitVolumeUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_volume_unit_of_measure_name: Option<String>,
    #[serde(rename = "ItemUnitWeight", skip_serializing_if = "Option::is_none")]
    pub item_unit_weight: Option<f64>,
    #[serde(rename = "ItemUnitWeightUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub item_unit_weight_unit_of_measure_name: Option<String>,
    #[serde(rename = "LabTestingState", skip_serializing_if = "Option::is_none")]
    pub lab_testing_state: Option<String>,
    #[serde(rename = "PackageId", skip_serializing_if = "Option::is_none")]
    pub package_id: Option<i64>,
    #[serde(rename = "PackageLabel", skip_serializing_if = "Option::is_none")]
    pub package_label: Option<String>,
    #[serde(rename = "PackageType", skip_serializing_if = "Option::is_none")]
    pub package_type: Option<String>,
    #[serde(rename = "PackagedDate", skip_serializing_if = "Option::is_none")]
    pub packaged_date: Option<String>,
    #[serde(rename = "ProductLabel", skip_serializing_if = "Option::is_none")]
    pub product_label: Option<TransfersTransferTemplateDeliveryPackageProductLabel>,
    #[serde(rename = "ProductRequiresRemediation", skip_serializing_if = "Option::is_none")]
    pub product_requires_remediation: Option<bool>,
    #[serde(rename = "ProductionBatchNumber", skip_serializing_if = "Option::is_none")]
    pub production_batch_number: Option<String>,
    #[serde(rename = "ReceivedQuantity", skip_serializing_if = "Option::is_none")]
    pub received_quantity: Option<f64>,
    #[serde(rename = "ReceivedUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub received_unit_of_measure_name: Option<String>,
    #[serde(rename = "RemediationDate", skip_serializing_if = "Option::is_none")]
    pub remediation_date: Option<String>,
    #[serde(rename = "SellByDate", skip_serializing_if = "Option::is_none")]
    pub sell_by_date: Option<String>,
    #[serde(rename = "ShipmentPackageState", skip_serializing_if = "Option::is_none")]
    pub shipment_package_state: Option<String>,
    #[serde(rename = "ShippedQuantity", skip_serializing_if = "Option::is_none")]
    pub shipped_quantity: Option<f64>,
    #[serde(rename = "ShippedUnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub shipped_unit_of_measure_name: Option<String>,
    #[serde(rename = "SourceHarvestNames", skip_serializing_if = "Option::is_none")]
    pub source_harvest_names: Option<String>,
    #[serde(rename = "SourcePackageIsDonation", skip_serializing_if = "Option::is_none")]
    pub source_package_is_donation: Option<bool>,
    #[serde(rename = "SourcePackageIsTradeSample", skip_serializing_if = "Option::is_none")]
    pub source_package_is_trade_sample: Option<bool>,
    #[serde(rename = "SourcePackageLabels", skip_serializing_if = "Option::is_none")]
    pub source_package_labels: Option<String>,
    #[serde(rename = "SourceProductionBatchNumbers", skip_serializing_if = "Option::is_none")]
    pub source_production_batch_numbers: Option<String>,
    #[serde(rename = "UseByDate", skip_serializing_if = "Option::is_none")]
    pub use_by_date: Option<String>,
}
